lib/logproto/logproto-text-server.c

```diff
@@ -199,7 +200,7 @@ log_proto_text_server_split_buffer(LogProtoTextServer *self, LogProtoBufferedSer
 
 
       if (self->super.super.options->encoding)
-        raw_split_size = log_proto_text_server_get_raw_size_of_buffer(self, buffer_start, buffer_bytes);
+        raw_split_size = log_proto_text_server_get_raw_size_of_buffer(self, *buffer_start, buffer_bytes);
```

Comment:
If I understand things right, then this is the only problem that we are fixing. log_proto_text_server_get_raw_size_of_buffer() needs the new buffer_start (e.g. the beginning of self->super.buffer) instead of the buffer we got as parameter. Is that right?

If so, then I don't think we need to change the argument to be a pointer to a pointer, the fix is simply using self->super.buffer as the 2nd argument to log_proto_text_server_get_raw_size_of_buffer()

Reply:
Yes, that is the only problematic misuse currently. If we don't change the signature, I'll have to add a comment to buffered server and to text server to indicate that the local variable should not be used below the problematic function call.